### ISSUE TITLE
`netrc` omits `login`, defaulting to system username

### ISSUE THREAD

**Issue by ivb-supercede** [cachix/cachix#406]
Because the `netrc` configuration does not specify a value for `login` ([here](https://github.com/cachix/cachix/blob/0295dac48d2857f497cf3a9b6467f399853743b7/cachix/src/Cachix/Client/NetRc.hs#L51)), it will use the fallback of the current username in the system.

This will cause an authentication error when using a cache where your Cachix username is not the same as your system username on the system you are trying to connect to Cachix with. This was an issue for me because my Cachix username comes from my GitHub account, and my system username is of course allowed to be whatever I like.

**Comment by domenkozar** [comment]
login value is ignored, so something else is going on 🤔

**Comment by ivb-supercede** [comment]
>login value is ignored

Not when testing with `curl`:
```
curl https://supercede.cachix.org/nix-cache-info --netrc-file ~/.config/nix/netrc
```
This would give a "does not exist" response until I specified the `login` in the `netrc`. Nix was also telling me that `supercede.cachix.org` did not appear to be a cache.

**Comment by domenkozar** [comment]
Note that this cache was renamed so it's probably why it gives you 404. I can't reproduce your problem locally.

**Comment by andreyk0** [comment]
FWIW adding `login` to `netrc` file just fixed this issue for me:

```
warning: 'https://<our-private-cache>.cachix.org/' does not appear to be a binary cache
```

Just re-installed `nix` from a pkg [these basic instructions](https://wiki.archlinux.org/title/Nix) , it got installed in a "daemon" mode.
Followed the usual "install `cachix`" steps after that and ended up with the above warning.
After reading this issue tried adding `login` and now I see it hit our private cache again. Maybe something changed with nix `2.8.0` release ? I don't remember seeing this before.

**Comment by domenkozar** [comment]
@ivb-supercede what's your nix version?

**Comment by ivb-supercede** [comment]
```
$> nix --version
nix (Nix) 2.6.1
```

**Comment by domenkozar** [comment]
I can confirm this is indeed a bug and just adding `login` to netrc resolves it. I'll hunt down the issue in Nix.

**Comment by dhess** [comment]
FYI, we just got hit by this and we're still running Nix 2.5.1.

**Comment by domenkozar** [comment]
So you just updated nixpkgs?

**Comment by dhess** [comment]
Yes, I would guess this has been affecting us for approximately the last 2 weeks. We bump `nixpkgs` to the latest `nixpkgs-unstable` roughly once a week.

**Comment by ivb-supercede** [comment]
In fact, we had just updated our nixpkgs as well before I reported this issue - it slipped my mind because I assumed it was instead related to the cache renaming which happened at a similar time.

**Comment by domenkozar** [comment]
This is a regression in curl: https://github.com/curl/curl/issues/8653

**Comment by domenkozar** [comment]
Fix is now pending for nixpkgs: https://github.com/NixOS/nixpkgs/pull/176974

**Comment by domenkozar** [comment]
This is now fixed in nixpkgs master and 22.05, hopefully channels will update tomorrow!

**Comment by domenkozar** [comment]
It's in the channels now and I confirmed it fixes the issue.

Thanks everyone for reporting and sorry for this terrible curl bug :(
